pkg/monitor/worker.go

```diff
@@ -18,6 +20,30 @@ import (
 	"github.com/Azure/ARO-RP/pkg/util/restconfig"
 )
 
+// This function will continue to run until such time as it has a config to add to the global Hive shard map
+// Note that because the mon.hiveShardConfigs[shard] is set to `nil` when its created, the cluster
+// monitors will simply ignore Hive stats until this function populates the config
+func (mon *monitor) populateHiveShardRestConfig(ctx context.Context, shard int) {
+	var hiveRestConfig *rest.Config
+	var err error
+
+	for {
+		hiveRestConfig, err = mon.liveConfig.HiveRestConfig(ctx, shard)
+		if hiveRestConfig != nil {
+			mon.setHiveShardConfig(shard, hiveRestConfig)
+			return
+		}
+
+		mon.baseLog.Warn(fmt.Sprintf("error fetching Hive kubeconfig for shard %d", shard))
+		if err != nil {
+			mon.baseLog.Error(err.Error())
+		}
+
+		mon.baseLog.Info("pausing for a minute before retrying...")
+		time.Sleep(time.Duration(60 * time.Second))
```

Comment:
`time.Sleep(60 * time.Second)` should be enough.
Moreover, I am tempted to prolong the interval to give AKS some rest. `2-5` minutes?

Reply:
I spent quite a bit of time deciding on the intervals actually and I think 60 seconds if a good compromise for not wanting to be "aggressive" and also needing to get the config quickly. We currently [poll the AAD endpoint every 10 seconds](https://github.com/Azure/ARO-RP/blob/f3d5f276bb2f07ec05d428aec1aaf575479d98fa/pkg/util/aad/aad.go#L46), when we are trying to get valid credentials, so 60 doesn't seem to high in this instance.